setup.py

```diff
@@ -1,9 +1,15 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
-from distutils.core import setup, Distribution
-from distutils.command.build_py import build_py
-from distutils.extension import Extension
-from Cython.Build import cythonize
+try:
+    from setuptools.command.build_ext import build_ext as build_ext
+    from setuptools import setup, Distribution, Extension
+except ImportError:
+    from distutils.command.build_ext import build_ext as build_ext
+    from distutils.core import setup, Distribution
+    from distutils.extension import Extension
```

Comment:
I really don't like these conditional imports. Either use `setuptools` or use `distutils` but I don't want to support both.